### PR TITLE
[Merged by Bors] - feat: use identity guard (PL-625)

### PR DESCRIPTION
**Pull request by trs**
Utilize standard identity guard, remove `verifyDMAPIKey`

### CODE REVIEW

File: lib/middlewares/auth.ts [hunk 1]
```diff
@@ -51,49 +38,59 @@ class Auth extends AbstractMiddleware {
...
+  };
 
-      if (!identity?.identity?.id) throw new Error();
+  verifyIdentity = async (req: Request, _res: Response, next: Next): Promise<void> => {
+    try {
+      const guard = await this.getIdentityGuard();
+      if (!guard) return next();
 
-      return next();
+      return guard()(req, res, next);
```
Comment (zhihil) [inline lib/middlewares/auth.ts:89]: Could we abstract the `getXGuard()` and `verifyX()` methods? Seems their overall logic is similar (e.g. `verifyIdentity()`, `verifyPermissions()`), but they differ only in their strategy, i.e, Identity or Permissions.

Comment (zhihil) [reply lib/middlewares/auth.ts:89]: Nevermind, if this is a temporary pattern, then it's not worth the abstraction.

**Review by pmvrmc — APPROVED** [review]
there is a build issue, but code looks good.

**Review by theProf — APPROVED** [review]

**Comment by trs** [comment]
bors merge

**Pull request closed without merging**